File: 1755-DefuseTheBomb/1755-DefuseTheBomb.js
// Last updated: 4/29/2026, 10:22:58 AM
/**
 * @param {number[]} code
 * @param {number} k
 * @return {number[]}
 */
var decrypt = function(code, k) {
    const n = code.length;
    const next = (i, k) => {
        if(k > 0) {
            return (i + 1) % n;    
        }
        return (i - 1 + n) % n;
    }
    const res = [];
    for(let i = 0; i < n; i++) {
        let sum = 0;
        let currIdx = i;
        for(let j = 0; j < Math.abs(k); j++) {
            currIdx = next(currIdx, k);
            sum += code[currIdx];
        }
        res.push(sum);
    }
    return res;
};

File: 1755-DefuseTheBomb/1755-DefuseTheBomb.py
# Last updated: 4/29/2026, 10:22:53 AM
class Solution:
    def decrypt(self, code: List[int], k: int) -> List[int]:
        n = len(code)
        
        def next(i, k):
            if k > 0:
                return (i+1)%n
            return (i-1+n)%n
        
        ret = []
        for i in range(n):
            sum = 0
            curr = i
            for j in range(abs(k)):
                curr = next(curr, k)
                sum += code[curr]
            ret.append(sum)
        
        return ret